x-pack/plugins/maps/public/classes/layers/blended_vector_layer/blended_vector_layer.ts

```diff
@@ -248,6 +248,10 @@ export class BlendedVectorLayer extends VectorLayer implements IVectorLayer {
   }
 
   async syncData(syncContext: SyncContext) {
+    if (!this.isVisible() || !this.showAtZoomLevel(syncContext.dataFilters.zoom)) {
```

Comment:
this reoccurs everywhere.

e.g.: 
- all layers, 
- map_selectors, 
- toc_entry, 


What about introducing a new method on ILayer, something like `ILayer.isVisibleAndWithinZoom(zoom: number):boolean` and just reuse?

Reply:
I would like to avoid to large of changes for the fix since this is getting backported to 7.8. How about keeping this PR to just the fix. 7.8 and master are not going to have a clean backport because layers folder was renamed to classes. 

Then, creating a separate PR that refactors and adds tests.

Reply:
talked offline, will move visibility check out of syncData and into redux action that call syncData so layers do not need to perform this check and check can be done in a single location.